server/mongo_data_access.go

```diff
@@ -14,16 +14,67 @@ import (
 	"gopkg.in/mgo.v2/bson"
 )
 
+// MongoConnection manages the direct connection to the Mongo database
+type MongoConnection struct {
+	session *mgo.Session
+	dbname  string
+}
+
+// Connect establishes a session with the mongo database
+func (dc *MongoConnection) Connect(host string) (err error) {
+	dc.session, err = mgo.Dial(host)
+	return
+}
+
+// SetDatabaseName sets the name of the database MongoConnection should use
+func (dc *MongoConnection) SetDatabaseName(dbname string) {
+	dc.dbname = dbname
+}
+
+// SetSession allows you to set the MongoConnection session to an existing session
+// object rather than using Connect. Currently this is just used for testing. SetSession
+// will fail if the MongoConnection already has an established session.
+func (dc *MongoConnection) SetSession(session *mgo.Session) (err error) {
+	if dc.session != nil {
+		return errors.New("MongoConnection: Cannot set session, session already established")
+	}
+	dc.session = session
+	return nil
+}
+
+// SetTimeout sets the session timeout for the current connection. The default timeout is 1 minute.
+func (dc *MongoConnection) SetTimeout(d time.Duration) {
+	dc.session.SetSocketTimeout(d)
+}
+
+// Copy returns a copy of the current MongoConnection and session
+func (dc *MongoConnection) Copy() *MongoConnection {
+	return &MongoConnection{
+		session: dc.session.Copy(),
+		dbname:  dc.dbname,
+	}
+}
+
+// Database returns a database accessible from the current session
+func (dc *MongoConnection) Database() *mgo.Database {
```

Comment:
`Database()` is kind of long.  Consider taking a cue from mgo (see line directly below) and just calling it `DB()`.  Less typing that way, which is good since this function gets called a fair bit.  That said, I wouldn't reject the PR just for this.